File: collatz/collatz.py
#!/usr/bin/env python

from sys import argv

LOG = False

def collatz(n):
	"""Test the collatz conjecture for some positive integer n"""
	if(LOG): print(n)
	if n == 1:
		print("Complete.")
		return
	if(n%2 == 0): # even
		collatz(n//2)
	else: # odd
		collatz(3*n+1)

def main():
	if(len(argv) != 2):
		n = int(input("Enter a number: "))
	else:
		n = int(argv[1])
	if(LOG): print(n)
	collatz(n)

if(__name__ == "__main__"):
	main()

File: collatz/collatz.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

#define LOGGING false

bool collatz(int n) {
	if(LOGGING) { printf("n = %d\n", n); }

	if(n == 1) {
		printf("Complete.\n");
		return true;
	}

	if(n % 2 == 0) { // even
		return collatz(n/2);
	} else { // odd
		return collatz(3*n+1);
	}
}

int main(int argc, char **argv) {
	if(argc != 2) {
		return 1; // error
	}

	int n = atoi(argv[1]);
	if(LOGGING) {printf("%d", n); }
	collatz(n);
	return 0;
}

void scan() {
	int n;
	scanf("%d", &n);
	collatz(n);
}
